Configure retry on Rate-Limiting from remote-write config (#8477)
prometheus/prometheus: docs/configuration/configuration.md

headers:
  [ <string>: <string> ... ]

> # Retry on receiving a rate-limiting status code 429, in response from remote-write storage.
> # This is expected to change at any point or be removed in future releases.

Reviewer: I'd suggest alternate wording, and explictely mark this as experimental.

```suggestion
# Retry upon receiving a 429 status code from the remote-write storage.
# This is expected to change at any point or be removed in future releases.
```

Author: Done

headers:
  [ <string>: <string> ... ]

> # Retry on receiving a rate-limiting status code in response from remote-write storage.

Reviewer 2: Would you put a note here saying that this configuration may change in any release, similar to the note above `metadata_config`. I imagine we may get rid of it once there is an agreed upon solution to how to handle rate limited retries.

Reviewer: HTTP 429 should be mentioned.

I am not confortable to change this behavior in a minor release.

Reviewer 2: This doc string should also say retry on 429 status code, and the variable name two lines down still needs to be updated.

Author: variable name?

Reviewer 2: `retry_on_rate_limit` should be `retry_on_http_429` now a couple lines below this.

Author: Done, I hope the message is good.
